packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -25,18 +30,37 @@ export * from './types';
 
 export const REQUEST_RATE = 10;
 
+if (process.env.REACT_APP_FAKE_CHANNEL_PROVIDER === 'true') {
+  window.channelProvider = new FakeChannelProvider();
+} else {
+  // TODO: Replace with injection via other means than direct app import
+  // NOTE: This adds `channelProvider` to the `Window` object
+  require('@statechannels/channel-provider');
+}
+
+// TODO: Put inside better place than here where app can handle error case
+window.channelProvider.enable(process.env.REACT_APP_WALLET_URL);
+
 export default class WebTorrentPaidStreamingClient extends WebTorrent {
   allowedPeers: PeersByTorrent;
   pseAccount: string;
   torrents: PaidStreamingTorrent[] = [];
+  channelClient: Web3TorrentChannelClientInterface;
+  amSeeder: boolean = false;
```

Comment:
This is probably a bad idea in the long term since I can become a seeder at any point.

- The idea of a leecher being "not a seeder" is not really correct.
- We still want the "downloading peer" to open the channel during the opening handshake, if we can deduce who that is.

Reply:
It keeps going back to this, isn't it?

The problem that I keep finding is that a peer can behave as a seeder in a moment, and as a leecher in the next, with another peer. As the wire is a duplex connection, request for parts of files can come from both peers.

I thought about two options to handle this:
- We create the channel, and make the peers join **not** on the handshake, but when the first request is received. That way, we have certainty about who's gonna fill which role.
We could do that on: https://github.com/statechannels/monorepo/blob/89202f6ae7e8bb8d96e0a41c771f446930b62874/packages/web3torrent/src/library/web3torrent-lib.ts#L129


- Another thing that I think it could be possible, depending on the channel's support, would be to drop all pretences of differentiating between seeders and leechers, and making that the balance can be positive or negative, or even maintaining 2 balances per channel, where each peer A checks the balance of peer B when a request is made, and vice-versa.
That way, it doesn't really matter who's the seeder, and who's the leecher, both peers join the channel in equal conditions.

Reply:
I like your first suggestion!

I think your suggestion would be quite elegant and match the true "peer-to-peer" nature of bittorrent, but  would be much more difficult given that ForceMove channels are turn-based. The way we currently handle the setup phase of a channel is really forcing us into having one proposer and one confirmer. 

As you say, whomever sends the first request can propose the channel. Otherwise, I can always accept channels that are being proposed (possible DoS vector here). 

If two peers are seeding and leeching from each other (which I think is possible), we can just have two unidirectional payment channels for that.